packages/labs/compiler/src/lib/template-transform.ts

```diff
@@ -68,12 +76,29 @@ class CompiledTemplatePass {
       return (sourceFile: ts.SourceFile): ts.SourceFile => {
         const pass = new CompiledTemplatePass(context, sourceFile);
         pass.findTemplates(sourceFile);
+        if (pass.expressionToTemplate.size === 0) {
+          // No templates to compile.
+          return sourceFile;
+        }
         const transformedSourceFile = pass.rewriteTemplates(sourceFile);
         if (!ts.isSourceFile(transformedSourceFile)) {
           throw new Error(
             `Internal Error: Expected source file to be transformed into another source file.`
           );
         }
+        if (pass.addedSecurityBrandVariableStatement === null) {
+          // No compilation occurred.
+          return sourceFile;
```

Comment:
this might lead to a confusing situation if later on we introduce changes do some transformation but don't always add the security brand statement for some reason

I think we should return transformedSourceFile here, it's the same as returning sourceFile isn't it, and this is a rare case anyways, where we found templates but bailed out of compiling them?

Reply:
Agree – done. I removed this and rolled the expectation of the security brand into the next `if` statement. That also simplifies the flow.